ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAlgorithm.scala

```diff
@@ -56,10 +56,11 @@ abstract class H2OAlgorithm[B <: H2OBaseModelBuilder : ClassTag, M <: H2OBaseMod
     val binaryModel: H2OBaseModel = trainModel(parameters)
     val mojoData = ModelSerializationSupport.getMojoData(binaryModel)
     val modelSettings = H2OMOJOSettings.createFromModelParams(this)
-    H2OMOJOModel.createFromMojo(
+    val model = H2OMOJOModel.createFromMojo(
       mojoData,
       Identifiable.randomUID(binaryModel._parms.algoName()),
       modelSettings)
+    model.set(model.featuresCols, getFeaturesCols())
```

Comment:
When the mojo is trained in Sparkling Water, we override the feature names with the non-expanded ones as the code later takes care of the column expansion.